src/Storages/MergeTree/MergeTreeDataSelectExecutor.cpp

```diff
@@ -1102,6 +1102,11 @@ std::shared_ptr<QueryIdHolder> MergeTreeDataSelectExecutor::checkLimits(
                 max_partitions_to_read);
     }
 
+// TSA is confused by guaranteed copy elision in data.getQueryIdSetLock()
+#ifdef __clang__
+#pragma clang diagnostic push
+#pragma clang diagnostic ignored "-Wthread-safety-analysis"
+#endif
```

Comment:
Is it possible to use TSA_NO_THREAD_SAFETY_ANALYSIS here? At least we will have some usage example in the codebase)